File: src/Back/Back.Domain/Models/Ingredient.cs
using Back.Domain.Abstractions;
using Back.Domain.ValueObjects;

namespace Back.Domain.Models
{
    public class Ingredient : Entity<IngredientId>
    {
        public string Name { get; private set; } = default!;
        public decimal Carbs { get; private set; } = default!;
        public decimal Sugar { get; private set; } = default!;
        public decimal Fat { get; private set; } = default!;
        public decimal Protein { get; private set; } = default!;
        public decimal kCal { get; private set; } = default!;
        public decimal GI { get; private set; } = default!; //GlygemicIndex

        public static Ingredient Create(IngredientId id, string name, decimal carbs, decimal sugar, decimal fat, decimal protein, decimal kcal, decimal gi)
        {
            ArgumentException.ThrowIfNullOrWhiteSpace(name);
            ArgumentOutOfRangeException.ThrowIfNegative(carbs);
            ArgumentOutOfRangeException.ThrowIfNegative(sugar);
            ArgumentOutOfRangeException.ThrowIfNegative(fat);
            ArgumentOutOfRangeException.ThrowIfNegative(protein);
            ArgumentOutOfRangeException.ThrowIfNegative(kcal);
            ArgumentOutOfRangeException.ThrowIfNegative(gi);

            var ingredient = new Ingredient
            {
                Id = id,
                Name = name,
                Carbs = carbs,
                Sugar = sugar,
                Fat = fat,
                Protein = protein,
                kCal = kcal,
                GI = gi
            };

            return ingredient;
        }
    }
}

File: src/backend-java/src/main/java/recepti_type1/backend_java/Models/Ingredient.java
package recepti_type1.backend_java.Models;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.util.UUID;

@Entity
@Table(name = "\"Ingredients\"")
public class Ingredient
{
    @Id
    private UUID Id;
    private String Name;
    private Float Carbs;
    private Float Sugar;
    private Float Fat;
    private Float Protein;
    private Float kCal;
    private Float GI; // Glycemic Index

    public Ingredient() {}

    public Ingredient(UUID id, String name, Float carbs, Float sugar,
                      Float fat, Float protein, Float kcal, Float gi)
    {
        this.Id = id;
        this.Name = name;
        this.Carbs = carbs;
        this.Sugar = sugar;
        this.Fat = fat;
        this.Protein = protein;
        this.kCal = kcal;
        this.GI = gi;
    }

    public UUID getId() {
        return Id;
    }

    public String getName() {
        return Name;
    }

    public Float getCarbs() {
        return Carbs;
    }

    public Float getSugar() {
        return Sugar;
    }

    public Float getFat() {
        return Fat;
    }

    public Float getProtein() {
        return Protein;
    }

    public Float getKcal() {
        return kCal;
    }

    public Float getGi() {
        return GI;
    }

    public static Ingredient create(UUID id, String name, Float carbs, Float sugar,
                                    Float fat, Float protein, Float kcal, Float gi)
    {
        return new Ingredient(id, name, carbs, sugar, fat, protein, kcal, gi);
    }
}
